46.md

```diff
@@ -1,98 +1,174 @@
+
+### Remote signer-side
+
+1. Remote signers manage the private key of the user that will be used for signing. There are many different remote signers out there (e.g. Nsecbunker, Amber, etc.) that operate in different ways and have different features.
+2. Remote signers wait for request events and then act.
+
+### Example flow for signing an event
+
+Coming soon...
+
+### Example Oauth-like flow to create a new user account with Nsecbunker
+
+Coming soon...
 
-## Event payloads
+## Request Events `Multiple kinds`
 
-Event payloads are [NIP-04](04.md)-encrypted JSON blobs that look like JSONRPC messages (their format is specified inside the `.content` of the event formats below).
+Request events take the following form. In order simplify requests and payloads as much as possible, there are different event kinds for each command which help remote signers to understand the request and process the content properly.
 
-Events sent by the client to the remote signer have the following format:
+Clients can subscribe to events of `kind: 24136`, `kind: 24137`, and `kind: 24138` that have an "e" tag containing the ID of their request event to listen for the response, auth_challenge, or error to their command.
 
-```js
+```json
 {
-  "pubkey": "<client-key-hex>"
-  "kind": 24133,
-  "tags": [
-    ["p", "<signer-key-hex>"]
-  ],
-  "content": "nip04_encrypted_json({id: <random-string>, method: <see-below>, params: [array_of_strings]})",
-  ...
+    "id": <id>,
+    "kind": <kind for command>, // Each command has it's own kind
+    "pubkey": <local pubkey of the user>,
+    "content": <nip44(<request>)>,
+    "tags": [["p", <remote pubkey>]],
+    "created_at": <unix timestamp in seconds>,
 }
 ```
 
-And the events the remote signer sends to the client have the following format:
+| Kind    | Command                  | Content field (always NIP-44 encrypted)                                         | Data Type               |
+| ------- | ------------------------ | ------------------------------------------------------------------------------- | ----------------------- |
+| `24140` | `create_account`         | {username: <username>, domain: <domain>, <any other arbitrary key/value pairs>} | JSON stringified object |
+| `24141` | `connect`                | -                                                                               | null                    |
+| `24142` | `sign_event`             | nostr event to sign                                                             | JSON stringified object |
+| `24143` | `ping`                   | -                                                                               | null                    |
+| `24144` | `get_relays`             | -                                                                               | null                    |
+| `24145` | `nip04_encrypt`          | {third_party_pubkey: <third_party_pubkey>, content: <plaintext to encrypt>}     | JSON stringified object |
```

Comment:
I would go with just `pubkey` for these methods. I think the directionality is inherently confusing (you have to ask, "third-party from the perspective of whom?"), and no possible name can help it, so I'd go with the simpler name, matching nostr-tools.

Reply:
the other option would be `recipient_pubkey` which is probably what I should have gone with in the first place. There are a lot of `pubkey`s in implementing NIP-46 so I think it's important to be as clear as possible, but maybe that's just a documentation thing. I'm happy with `recipient_pubkey` or just plain `pubkey`

Reply:
`recipient_pubkey` still has the directionality problem. If I had to use an underscored name I'd call it `encryption_pubkey`. But it's ugly enough I'd still rather just call it `pubkey`. I think you made it clear enough in every other part of the document. It's actually clearer to be less clear here.

Reply:
```suggestion
| `24145` | `nip04_encrypt`          | {pubkey: <third_party_pubkey>, content: <plaintext to encrypt>}     | JSON stringified object |
```
I think the key should just be `pubkey` but the placeholder in brackets should be descriptive like you have it